src/Cli/Microsoft.TemplateEngine.Cli/Commands/create/InstantiateCommand.cs

```diff
@@ -172,6 +173,26 @@ async Task<string> GetTemplatePackagesList(TemplateGroup templateGroup)
             }
         }
 
+        // Reparses the CLI parameters in the context of the provided command (usually TemplateCommand).
+        // If the first token after reparsing starts with a dash, we reparse again and inject a name sentinel to allow it to be handled properly (using empty name logic).
+        internal static ParseResult Reparse(CliCommand command, IReadOnlyList<string> args, CliConfiguration? configuration = null)
+        {
+            var argsList = args.ToList();
+            ParseResult parseResult = command.Parse(args, configuration);
+
+#pragma warning disable CA1826
+            var firstTemplateCommandToken = parseResult.CommandResult.Tokens.FirstOrDefault()?.Value ?? string.Empty;
+#pragma warning restore CA1826
+            int tokenIndex;
+            if (firstTemplateCommandToken.StartsWith('-') && (tokenIndex = argsList.IndexOf(firstTemplateCommandToken)) >= 0)
+            {
+                argsList.Insert(tokenIndex, TemplateCommandArgs.NameDefaultSentinel);
+                parseResult = command.Parse(argsList, configuration);
```

Comment:
This solution does require this second parse as you need to have the tokens individually first; then use the sentinel if the first token isn't supposed to be `name`.